pkg/cmd/issue/edit/edit.go

```diff
@@ -0,0 +1,236 @@
+package edit
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	shared "github.com/cli/cli/pkg/cmd/issue/shared"
+	prShared "github.com/cli/cli/pkg/cmd/pr/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/shurcooL/githubv4"
+	"github.com/spf13/cobra"
+)
+
+type EditOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	OpenInBrowser      func(string) error
+	DetermineEditor    func() (string, error)
+	FieldsToEditSurvey func(*prShared.EditableOptions) error
+	EditableSurvey     func(string, *prShared.EditableOptions) error
+	FetchOptions       func(*api.Client, ghrepo.Interface, *prShared.EditableOptions) error
```

Comment:
Instead of passing around funcs for the purpose of stubbing in tests—even though that is the pattern that we practiced before—how about we consider [gradually moving to interfaces](https://gist.github.com/mislav/eb956ff0f03fb8fc255ab3aa72be7113)? It would allow the same kind of dependency injection, with the added benefit of better readability/being more idiomatic Go.

You don't have to rewrite all of these; I'm just wondering whether you think that accepting interfaces and then substituting the concrete implementation object in tests would be an approach that could make the above cleaner.

Reply:
I like the interface approach and generally think it is better. I mostly just forgot that we had discussed it when I wrote this and ended up falling back on what we had been doing. 

My only real complaint with the interface approach is that it doesn't seem to bring much improvement when the interface defines only one function and results in slightly more code. I do think bundling up the survey functions into a `surveyAsker` interface, as seen in the gist, is great though. 

I will keep the idea top of mind when I do the `pr edit` command and see how that goes, and if it works well I will come back and update this command.